File: Python/5 Kyu/simplePigLatin.py
"""
Move the first letter of each word to the end of it, then add "ay" to the end of the word. Leave punctuation marks untouched.

Examples
pig_it('Pig latin is cool') # igPay atinlay siay oolcay
pig_it('Hello world !')     # elloHay orldway !
"""


#MY SOLUTION
def pig_it(text):
    letters = "abcdefghijklmnopqrstuvwxyz"
    return_list = text.split(' ')
    text_list = []
    temp = ""
    tempWord = ""
    for word in return_list:
        if word[0].lower() in letters:
            temp = word[0]
            tempWord = word[1:]
            tempWord += temp + "ay"
            text_list.append(tempWord)
        else:
            text_list.append(word)
    return (' ').join(text_list)
            

File: JavaScript/5 Kyu/simplePigLatin.js
/*
Move the first letter of each word to the end of it, then add "ay" to the end of the word. Leave punctuation marks untouched.

Examples
pigIt('Pig latin is cool'); // igPay atinlay siay oolcay
pigIt('Hello world !');     // elloHay orldway !
*/
// MY SOLUTION

const pigIt = (str) => {
    let string = str.split(' ').map(x=>{
        let thisBool = true
        for (let i = 0; i < x.length; i++){
            if(!'abcdefghijklmnopqrstuvwxyz'.includes(x[i].toLowerCase())){
                thisBool = false
            }
        }
        if (thisBool){
            return x.slice(1) + x[0] + 'ay'
        }
        return x
    })
    return string.join(' ')
}